src/java.base/share/classes/java/util/zip/Inflater.java

```diff
@@ -59,8 +59,10 @@
  * {@snippet id="compdecomp" lang="java" class="Snippets" region="DeflaterInflaterExample"}
  *
  * @apiNote
- * To release resources used by this {@code Inflater}, the {@link #end()} method
- * should be called explicitly. Subclasses are responsible for the cleanup of resources
+ * The {@code Inflater} cleans up its resources in its {@link #end()} method. To facilitate
+ * usage with try-with-resources, this class implements {@link AutoCloseable}. The
+ * {@link #close()} method of this class calls {@code end()} to clean up its
+ * resources. Subclasses are responsible for the cleanup of resources
```

Comment:
Once we've agreed on the arrangement in the Deflater class, the corresponding changes should be made to Inflater, mutatis mutandis.